File: frontend/src/types/agent.ts
export interface Message {
  id?: number
  role: 'user' | 'assistant' | 'system' | 'tool' | 'thought'
  content: string
  tool_calls?: ToolCall[]
  token_usage?: TokenUsage
  created_at?: string
  step?: number
}

export interface ToolCall {
  name: string
  arguments: Record<string, any>
  result?: any
  status?: 'running' | 'success' | 'error'
  duration_ms?: number
  error?: string
}

export interface ThinkingStep {
  step: number
  thought?: string
  type: 'thinking' | 'tool_call' | 'tool_result'
  tool?: string
  arguments?: Record<string, any>
  result?: any
  duration_ms?: number
  error?: string
  status?: 'running' | 'success' | 'error'
  success?: boolean
}

export interface TokenUsage {
  prompt_tokens?: number
  completion_tokens?: number
  total_tokens?: number
}

export interface Source {
  file_name: string
  page_number?: number | string
  snippet?: string
}

export interface ChatSession {
  id: string
  title: string
  mode: string
  created_at: string
  updated_at: string
  message_count: number
  summary?: string
}

export interface ChatSessionDetail extends ChatSession {
  messages: Message[]
}

export interface ToolSchema {
  name: string
  description: string
  parameters: Record<string, any>
}

export interface SyncChatResponse {
  session_id: string
  message_id: number
  content: string
  steps: number
  tools_used: string[]
  sources: Source[]
  usage?: TokenUsage
  duration_ms: number
}

export interface SSEEvent {
  event: 'metadata' | 'thinking' | 'tool_call' | 'tool_result' | 'token' | 'done' | 'error'
  data: any
}

File: backend/app/schemas/agent.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import datetime


class AgentChatRequest(BaseModel):
    message: str = Field(..., description="用户消息")
    session_id: Optional[str] = Field(None, description="会话ID，不传则新建会话")
    user_name: str = Field("default_user", description="用户名")


class AgentChatSyncResponse(BaseModel):
    session_id: str
    message_id: int
    content: str
    steps: int
    tools_used: List[str] = []
    sources: List[Dict] = []
    usage: Optional[Dict] = None
    duration_ms: int


class SessionResponse(BaseModel):
    id: str
    title: str
    mode: str
    created_at: datetime
    updated_at: datetime
    message_count: int
    summary: Optional[str] = ""


class SessionListResponse(BaseModel):
    sessions: List[SessionResponse]
    total: int


class SessionDetailResponse(SessionResponse):
    messages: List[Dict[str, Any]]


class ToolSchema(BaseModel):
    name: str
    description: str
    parameters: Dict[str, Any]


class ToolListResponse(BaseModel):
    tools: List[ToolSchema]
    total: int


class ToolInvokeRequest(BaseModel):
    arguments: Dict[str, Any] = Field(default_factory=dict)


class ToolInvokeResponse(BaseModel):
    tool: str
    success: bool
    data: Any = None
    error: Optional[str] = None
    duration_ms: int
